lib/diplomat/key.ex

```diff
@@ -143,29 +129,22 @@ defmodule Diplomat.Key do
   @spec complete?(t) :: boolean
   def complete?(%__MODULE__{}=k), do: !incomplete?(k)
 
-  defp ancestors_and_self(nil, acc), do: Enum.reverse(acc)
-  defp ancestors_and_self(key, acc) do
-    ancestors_and_self(key.parent, [key|acc])
-  end
-
-  defp generate_path([], acc), do: acc
-  defp generate_path([key|tail], acc) do
-    generate_path tail, [[key.kind, (key.id || key.name)] | acc]
-  end
-
-  def from_allocate_ids_proto(%Diplomat.Proto.AllocateIdsResponse{keys: keys}) do
+  @spec from_allocate_ids_proto(AllocateIdsResponse.t) :: [t]
+  def from_allocate_ids_proto(%AllocateIdsResponse{keys: keys}) do
     keys |> Enum.map(&from_proto(&1))
   end
 
+  @spec allocate_ids(String.t, pos_integer) :: [t] | Client.error
   def allocate_ids(type, count \\ 1) do
-    keys = Enum.map 1..count, fn(_i) ->
-      __MODULE__.new(type) |> __MODULE__.proto
+    keys = Enum.map 1..count, fn _ ->
+      type |> new() |> proto()
```

Comment:
No idea why I was doing silly things like `__MODULE__.new` and `__MODULE__.proto`...